### PR TITLE
[RFC] Allow directives on variable definitions

### PR DESCRIPTION
**Pull request by mjmahone**
This implements directives on variable definitions, as defined in the [GraphQL RFC](https://github.com/facebook/graphql/pull/486)

### CODE REVIEW

**Comment by IvanGoncharov** [comment]
@mjmahone Also need to adjust this function:
https://github.com/graphql/graphql-js/blob/e124ca870a57ee85cf73a77863e69889bc11737d/src/validation/rules/KnownDirectives.js#L63

**Comment by mjmahone** [comment]
Thanks @IvanGoncharov: looks like I missed a few other locations as well, such as the introspection query. Added them just now.

**Comment by mjmahone** [comment]
#1438 is strictly better than this PR, so closing.

**Comment by mjmahone** [comment]
Sorry I closed the wrong one.

**Comment by mjmahone** [comment]
Merged https://github.com/facebook/graphql/pull/486, so I'm also merging this.

**Comment by leebyron** [comment]
Please place this behind an experimental flag until we can go through the RFC process for the spec

File: src/language/visitor.js [hunk 1]
```diff
@@ -64,7 +64,7 @@ export const QueryDocumentKeys = {
     'directives',
     'selectionSet',
   ],
-  VariableDefinition: ['variable', 'type', 'defaultValue'],
+  VariableDefinition: ['variable', 'type', 'defaultValue', 'directives'],
```
Comment (leebyron) [inline src/language/visitor.js]: These typically follow the order found in the grammar, but this seems inconsistent

Comment (leebyron) [reply src/language/visitor.js]: My mistake. Consistent for this PR but inconsistent with the spec RFC

**Comment by mjmahone** [comment]
@leebyron will do. Working on that PR now.

**Comment by mjmahone** [comment]
#1454 puts this all behind an experimental flag.

**Pull request merged**